src/saltext/vmware/states/vmc_sddc.py

```diff
@@ -0,0 +1,397 @@
+    deployment_type=None,
+    host_instance_type=None,
+    msft_license_config=None,
+    sddc_id=None,
+    sddc_template_id=None,
+    sddc_type=None,
+    size=None,
+    skip_creating_vxlan=False,
+    sso_domain=None,
+    storage_capacity=None,
+    vpc_cidr=None,
+    vxlan_subnet=None,
+    validate_only=False,
+    verify_ssl=True,
+    cert=None,
+):
+    """
+    Ensure a given SDDC exists for given organization.
+
+    name
+        The name of SDDC that will be assigned to new created SDDC.
+
+    hostname
+        The host name of VMC.
+
+    refresh_key
+        API Token of the user which is used to get the Access Token required for VMC operations
+
+    authorization_host
+        Hostname of the VMC cloud console.
+
+    org_id
+        The ID of organization to which the SDDC belongs to.
+
+    num_hosts
+        An integer value to indicate number of hosts in a SDDC.
+
+    provider: String
+        Determines what additional properties are available based on cloud provider.
+        Possible values are: AWS, ZEROCLOUD
+
+    region: String
+        Aws region on which SDDC will be deployed.
+
+    account_link_config
+        (Optional) The account linking configuration, we will keep this one and remove accountLinkSddcConfig finally.
+
+        account_link_config expects to be passed as a dict having delay_account_link parameter
+            delay_account_link: Boolean
+            (Optional) Boolean flag identifying whether account linking should be delayed or not for the SDDC.
+
+            .. code::
+
+                {
+                    "delay_account_link": false
+                }
+
+    account_link_sddc_config
+        (Optional) A list of the AccountLinkSddcConfig which indicates SDDC linking configurations to use.
+
+        AccountLinkSddcConfig has two parameters connected_account_id  and customer_subnet_ids
+
+        'connected_account_id': String
+            (Optional) The ID of the customer connected account to work with.
+
+        'customer_subnet_ids': Array of String
+            (Optional) List of subnet Ids
+
+    deployment_type: String
+        (Optional) Denotes if request is for a SingleAZ or a MultiAZ SDDC. Default is SingleAZ.
+        Possible values are: SingleAZ , MultiAZ
+
+    host_instance_type: String
+        (Optional) The instance type for the esx hosts in the primary cluster of the SDDC.
+        Possible values are: i3.metal, r5.metal, i3en.metal
+
+    msft_license_config : MsftLicensingConfig
+        (Optional) Indicates the desired licensing support, if any, of Microsoft software.
+
+    sddc_id: String As UUID
+        (Optional)If provided, will be assigned as SDDC id of the provisioned SDDC.
+
+    sddc_template_id : String As UUID
+        (Optional) If provided, configuration from the template will applied to the provisioned SDDC.
+
+    sddc_type: String
+        (Optional)Denotes the sddc type , if the value is null or empty, the type is considered as default.
```

Comment:
**suggestion** parameters should be lower case, `SDDC type` is appropriate for the documentation prose. Also there's an extra space `type , ` -> `type, `

Reply:
Updated